js/app/widgets/dynamicLogo.js

```diff
@@ -139,6 +147,17 @@ var DynamicLogo = new Knowledge.Class({
         return this._mode;
     },
 
+    set layout(value) {
+        if (this._layout === value)
+            return;
+        this._layout = value;
+        this.queue_draw();
```

Comment:
This should actually call queue_resize() instead of draw

Reply:
This is a weird widget, it receives an imposed size and it will always make sure that everything fits inside the size. Therefore, changing the layout won't need a resize, because the widget will simply draw in whatever space has available.